spec/01-unit/21-dns-client/02-client_spec.lua

```diff
@@ -1506,19 +1506,23 @@ describe("[DNS client]", function()
       assert(client.init({
         noSynchronisation = false,
         order = { "A" },
+        search = {},
       }))
 
       local callcount = 0
       query_func = function(self, original_query_func, name, options)
         callcount = callcount + 1
-        return original_query_func(self, name, options)
+        -- ensure individual_toip always triggers a DNS query to avoid it
+        -- triggering only once due to a cache hit.
+        ngx.sleep(1)
```

Comment:
Why add this `ngx.sleep` and why it is 1 second? Could we get more explainations?

Reply:
`sleep()` introduces a network interval, without it, this query will get the answer directly and save it into cache, then other concurrent `resolve()` calling in thread will hit the cache.
In such case, we could not test the `individual` query from `individual_toip()` call, 0~1s is ok then

Reply:
From my understanding, in other word, the sleep here is simulating a network delay to ensure it can be verified that multiple calls of individual_toip can initiate dns query only once for the same domain.